server/src/main/java/org/apache/iotdb/db/utils/datastructure/RandomAccessArrayDeque.java

```diff
@@ -0,0 +1,939 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iotdb.db.utils.datastructure;
+
+import java.io.Serializable;
+import java.util.AbstractCollection;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.ConcurrentModificationException;
+import java.util.Deque;
+import java.util.Iterator;
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.Objects;
+import java.util.Queue;
+import java.util.RandomAccess;
+import java.util.Spliterator;
+import java.util.function.Consumer;
+
+/**
+ * RandomAccessArrayDeque is a modified version of ArrayDeque from java.util, providing the ability
+ * to access an element by index.
+ * Most codes are copied directly because the fields are package-private and cannot be accessed
+ * by a simple extension.
+ * @param <E>
+ */
+public class RandomAccessArrayDeque<E> extends AbstractCollection<E>
```

Comment:
Wow~ That's a fantastic structure but without test. May be we should add some test for it.

Reply:
Since the class is mainly copied from java.util, I think most of it is trustable, and I only add some tests of `get()`.